extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -171,6 +181,20 @@ public boolean isAlreadyExistsException(SQLException e) {
   }
 
   private Connection borrowConnection() throws SQLException {
-    return datasource.getConnection();
+    Connection connection = datasource.getConnection();
+    if (connection
+        .getMetaData()
```

Comment:
Isn't it expensive to do this on every `borrowConnection()`?

Reply:
let me capture this in a variable so that we can do it once per datasource.